src/routes/event/add.tsx

```diff
@@ -0,0 +1,103 @@
+// WIP
+import { Button, Container, Stack, TextField, Typography } from "@mui/material";
+import { onAuthStateChanged } from "firebase/auth";
+import { useNavigate } from "react-router-dom";
+import { auth } from "../../firebase";
+import AutoComplete, { createFilterOptions } from "@mui/material/Autocomplete";
+import * as React from "react";
+
+const filter = createFilterOptions<string>();
+
+export default function EventAdd() {
+  const navigate = useNavigate();
+  onAuthStateChanged(auth, (user) => {
+    if (!user) {
+      console.log("No user signed in");
+      navigate(-1);
+    }
+  });
```

Comment:
Why this is bad ?
https://www.freecodecamp.org/news/use-firebase-authentication-in-a-react-app/
scroll down until you see how it should be used with 'onAuthStateChanged' approach.